Build failure: neomutt

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix run github:NixOS/nixpkgs/master#neomutt` on darwin aarch64

### Build log
see https://github.com/NixOS/nixpkgs/runs/13454777089

### Additional context

Since https://github.com/NixOS/nixpkgs/pull/231638 neomutt is broken on darwin-aarch64. See the CI output: https://github.com/NixOS/nixpkgs/runs/13454777089

Tangent: is it normal that a failing CI build wasn't marked as a failed check, but rather just ignored?

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.4.0, macOS 13.3.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.3`
 - channels(user): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/hnvgbhbvd5mq8fzppw22ifrcnngd3fqc-source`
```

I can't install neomutt on MacOS with home-manager too
```
ld: warning: directory not found for option '-L/nix/store/31zl3spxz18c13kr068a49yx2w07vwbw-neomutt-20230512/lib'
ld: warning: directory not found for option '-L/nix/store/31zl3spxz18c13kr068a49yx2w07vwbw-neomutt-20230512/lib'
ld: warning: directory not found for option '-L/nix/store/31zl3spxz18c13kr068a49yx2w07vwbw-neomutt-20230512/lib'
ld: warning: directory not found for option '-L/nix/store/31zl3spxz18c13kr068a49yx2w07vwbw-neomutt-20230512/lib'
ld: warning: directory not found for option '-L/nix/store/31zl3spxz18c13kr068a49yx2w07vwbw-neomutt-20230512/lib'
ld: warning: directory not found for option '-L/nix/store/31zl3spxz18c13kr068a49yx2w07vwbw-neomutt-20230512/lib'
Undefined symbols for architecture x86_64:
  "_fmemopen", referenced from:
      _test_mutt_convert_file_to in mutt_convert_file_to.o
      _test_mutt_convert_file_from_to in mutt_convert_file_from_to.o
      _test_mutt_rfc822_read_line in mutt_rfc822_read_line.o
ld: symbol(s) not found for architecture x86_64
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:1709: test/neomutt-test] Error 1
error: builder for '/nix/store/md05gx1j52nlb3wi8xyvrjxfwc98glma-neomutt-20230512.drv' failed with exit code 2;
       last 10 log lines:
       > ld: warning: directory not found for option '-L/nix/store/31zl3spxz18c13kr068a49yx2w07vwbw-neomutt-20230512/lib'
       > ld: warning: directory not found for option '-L/nix/store/31zl3spxz18c13kr068a49yx2w07vwbw-neomutt-20230512/lib'
       > Undefined symbols for architecture x86_64:
       >   "_fmemopen", referenced from:
       >       _test_mutt_convert_file_to in mutt_convert_file_to.o
       >       _test_mutt_convert_file_from_to in mutt_convert_file_from_to.o
       >       _test_mutt_rfc822_read_line in mutt_rfc822_read_line.o
       > ld: symbol(s) not found for architecture x86_64
       > clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       > make: *** [Makefile:1709: test/neomutt-test] Error 1
       For full logs, run 'nix-store -l /nix/store/md05gx1j52nlb3wi8xyvrjxfwc98glma-neomutt-20230512.drv'.
```

--- a/pkgs/applications/networking/mailreaders/neomutt/default.nix
+++ b/pkgs/applications/networking/mailreaders/neomutt/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, gettext, makeWrapper, tcl, which
+{ lib, stdenv, fetchFromGitHub, fetchpatch, gettext, makeWrapper, tcl, which
 , ncurses, perl , cyrus_sasl, gss, gpgme, libkrb5, libidn2, libxml2, notmuch, openssl
 , lua, lmdb, libxslt, docbook_xsl, docbook_xml_dtd_42, w3m, mailcap, sqlite, zlib, lndir
 , pkg-config, zstd, enableZstd ? true, enableMixmaster ? false, enableLua ? false
@@ -19,6 +19,12 @@ stdenv.mkDerivation rec {
   patches = [
     # https://github.com/neomutt/neomutt/issues/3773#issuecomment-1493295144
     ./fix-open-very-large-mailbox.patch
+    (fetchpatch {
+      name = "disable-incorrect-tests.patch";
+      url = "https://github.com/neomutt/neomutt/pull/3933.patch";
+      hash = "sha256-Plei063T8XyXF4/7/nAb6/4OyXz72vBAXHwls9WL1vM=";
+      excludes = [".github/workflows/macos.yml"];
+    })
   ];
 
   buildInputs = [

